lib/ui/painting.dart

```diff
@@ -42,6 +42,9 @@ bool _offsetIsValid(Offset offset) {
 bool _matrix4IsValid(Float64List matrix4) {
   assert(matrix4 != null, 'Matrix4 argument was null.');
   assert(matrix4.length == 16, 'Matrix4 must have 16 entries.');
+  assert(() {
+    return matrix4.every((double value) => value.isFinite);
```

Comment:
No need to wrap this in a closure - it'll get compiled out in AOT mode regardless.